src/regdesk/states/reg_machine_manager.js

```diff
@@ -116,4 +129,28 @@ export class RegMachineManager {
     this.#currentState = nextStateName;
     return this.#currentState;
   }
+
+  /**
+   * Determine if a cancel should go back to a multiple result page or the new search page.
+   * @return {string} - The next state to switch to.
+   */
+  #maybeCancelSingleResultToMultipleResult() {
+    if (this.#regMachineArgs.commManager.hasMultipleResults) {
```

Comment:
>  back button (which we should support imho, but not going to push it)

I'm going to veto this purely because the interface is designed to be used in fullscreen and the mice we're using don't have forward/back buttons. Could be a future improvement though, but definitely not for this version.

Rest of your comment is good feedback, going to take a look at a state chain. I dismissed it pretty early as I wanted entry into a state to be mostly idempotent but after some evolution I don't think that matters at this point.

Reply:
So we chatted a bit about the stack design and how we could implement it, and I think the call to be made here is that the system won't benefit from the added complexity of that design.

Right now we only have a handful of cancel 'targets', that is, states where you can end up after a cancel action happens. Those states are the 'multiple select state' and the 'new search' state. All other states are transient, in that you should not see them if you hit cancel.

This means the actual contents of a 'cancel stack' are going to be very limited, and we will need to describe that relationship as a _separate_ value within each state class. This functionally would mean two parallel state transition maps: one for forward workflows, one for back workflows. This also makes 'forward' operations much more complicated, as there are specific rules about whether a state can be re-entered after exiting it. 

I think this adds _power_, however that power would not be usefully employed for the con next week. If that makes sense? A todo for version 2.

Reply:
> Oh man, I thought we talked about law of demeter?

For this one specifically I don't think it applies. The regMachineArgs is a handy dandy transparent grouping of related arguments, not a class with behaviors unto itself, while the hasMultipleResults is an explicit item that commManager exposes.

Reply:
I can agree that a full stack is probably overpowered. But I think my argument that we shouldn't have a centralized side-effect storage class that determines state transitions still stands. But maybe you're responding to that in my other comment, so I'll sit on my hands for a bit. :)

Reply:
I think I'm going to call it as the PR currently stands at 9333c2b,

I've hauled the centralized state out and spread it among the various state elements, so this reduces the depth of information retrieval for the state transition determination method.

Right now the RegMachineManager is in charge of transitions between the states that it manages. I think it's fair game to say that the manager can introspect the various states under its management to determine how it should be redirecting events between the states. I don't think our complexity needs are going to grow to the point that we need another layer of abstraction on top of this, and I can already see how to fairly readily add in support for the SingleResultAlreadyPrinted state that we discussed adding today.

So I think I'm going to leave this specific bit of feedback for post-con cleanup, especially once future plans are solidified and we decide how much of this we're going to re-use.